src/view.cpp

```diff
@@ -247,7 +248,7 @@ int View::run()
 	return EXIT_SUCCESS;
 }
 
-std::string View::run_modal(std::shared_ptr<FormAction> f,
+nonstd::optional<std::string> View::run_modal(std::shared_ptr<FormAction> f,
```

Comment:
This method runs a formaction and returns a property named `value`; it can't distinguish between "no value" and "empty value", so I think it should keep returning `std::string`. The users of this method would have to process that and return `nullopt` as necessary.

Reply:
https://github.com/newsboat/newsboat/pull/2593/commits/bb6a768ef164eccc9adc4279fc913bab66d90936

Reply:
Okay, this takes care of the first part ("return `std::string`"), but the second part is still missing: `run_filebrowser()` and `run_dirbrowser()` should examine the result of `run_modal()` and return `nullopt` if it's an empty string.

Reply:
https://github.com/newsboat/newsboat/pull/2593/commits/28b628fc990060f8caac63eaf1ae492b106ffd76